src/main/java/io/github/ImpactDevelopment/installer/gui/pages/MainPage.java

```diff
@@ -27,27 +27,49 @@
 import io.github.ImpactDevelopment.installer.setting.InstallationConfig;
 import io.github.ImpactDevelopment.installer.setting.Setting;
 import io.github.ImpactDevelopment.installer.setting.settings.*;
+import io.github.ImpactDevelopment.installer.target.InstallationModeOptions;
 
 import javax.swing.*;
 import java.awt.*;
+import java.awt.event.ActionListener;
+import java.awt.event.KeyEvent;
 import java.net.URI;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 
+import static javax.swing.JOptionPane.*;
+
 public class MainPage extends JPanel {
     public MainPage(AppWindow app) {
         setLayout(new BoxLayout(this, BoxLayout.Y_AXIS));
-        addPathSetting(MinecraftDirectorySetting.INSTANCE, "Minecraft directory", app);
+        add(buildPathSetting(MinecraftDirectorySetting.INSTANCE, "Minecraft directory", JFileChooser.DIRECTORIES_ONLY, app));
         addSetting(InstallationModeSetting.INSTANCE, "Install for", app);
         addSetting(MinecraftVersionSetting.INSTANCE, "Minecraft version", app);
         addSetting(ImpactVersionSetting.INSTANCE, "Impact version", app);
-        addSetting(OptiFineSetting.INSTANCE, "OptiFine version", app);
+        if (!app.config.getSettingValue(InstallationModeSetting.INSTANCE).equals(InstallationModeOptions.FORGE)) {
```

Comment:
sorry to troll you like this, but i added a second FORGE type on master

Reply:
already spotted that, fixed when merging in master